src/components/list.jsx

```diff
@@ -10,57 +11,55 @@ class List extends Component {
         this.changeName = this.changeName.bind(this)
         this.handleKeyPress = this.handleKeyPress.bind(this)
         this.saveEdit = this.saveEdit.bind(this)
-        this.setEditMode = this.setEditMode.bind(this)
+        this.setTaskEditMode = this.setTaskEditMode.bind(this)
+        this.deleteList = this.deleteList.bind(this)
 
 
         this.state = {
-            listName: props.catalog.name,
+            listName: props.listItem.name,
             editMode: false,
         }
     }
 
-
     changeName(e) {
         this.setState({listName: e.target.value})
     }
 
     handleKeyPress(e) {
         if (e.key === 'Enter') {
-            console.log("enter key pressed")
             this.saveEdit()
         }
     }
 
     saveEdit() {
-        console.log("save - update the list name")
+        this.props.updateListName(this.props.currentListId, this.state.listName)
+        this.setTaskEditMode(false)()
     }
 
-    setEditMode(bool) {
+    setTaskEditMode(bool) {
         return () => this.setState({editMode: bool})
```

Comment:
why does this need to return a function? shouldn't this just call setstate with the supplied bool argument? also - variable should be descriptive of what it is (rather than its type)